web/src/components/GridCell.js

```diff
@@ -0,0 +1,114 @@
+
+class SingleGridCell extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      expanded: false,
+      selected_id: '',
+      window_width: window.innerWidth,
+      repoSelector: false,
+      slackInfo: false,
+      imageInfo: false,
+    }
+  }
+
+  cellClick(event) {
+    this.props.handleCellClick(event)
+  }
+
+  slackClicked = () => {
+    this.setState({slackInfo: true})
+  }
+
+  slackClosed = () => {
+    this.setState({slackInfo: false})
+  }
+
+  dockerClicked = () => {
+    this.setState({imageInfo: true})
+  }
+
+  dockerClosed = () => {
+    this.setState({imageInfo: false})
+  }
+
+  render() {
+    var SingleGridCellStyle = {
+      backgroundSize: this.props.cellSize,
+      width: this.props.cellSize,
+      height: this.props.cellSize,
+      display: 'inline-block',
+      margin: this.props.cellMargin,
+      marginBottom: 25,
+      position: 'relative'
+    }
+
+    var cardStyle = {
+      width: this.props.cellSize,
+      height: this.props.cellSize
+    }
+
+    // Re written to put material ui components in the tile original component only took images
+    var deployDate = new Date(this.props.SingleGridCellData.lastDeployed)
+    return (
+      <div style={SingleGridCellStyle} id={this.props.id} className='SingleGridCell'>
+        <div>
+          <Card style={cardStyle}>
+            <CardContent>
+              <Typography variant="h5" component="h2">
+                {this.props.SingleGridCellData.name}
+              </Typography>
+              <StatusChip status={this.props.SingleGridCellData.deployment.status} />
+              <div>
+                <IconButton>
+                  <img src={TimePassed} alt={imgAlt} />
+                </IconButton>
+                {deployDate.toDateString()}
+              </div>
+              <div className='row-align'>
+                <SelectionDialog gitref={this.props.SingleGridCellData.code.ref} highlanderPath={this.props.SingleGridCellData.code.github} serviceName={this.props.SingleGridCellData.name}/>
+                <IconButton onClick={this.slackClicked}>
+                  <img src={SlackIcon} width="32" height="32" alt={imgAlt} />
+                </IconButton>
+                {
+                  this.state.slackInfo ? <SlackInfoDialog open={this.state.slackInfo} owner={this.props.SingleGridCellData.owner.team} slack={this.props.SingleGridCellData.owner.slack} handleClose={this.slackClosed}/> : null
+                }
+                <IconButton onClick={this.dockerClicked}>
+                  <img src={DockerIcon} width="32" height="32" alt={imgAlt} />
+                </IconButton>
+                {
+                  this.state.imageInfo ? <ImageInfoDialog open={this.state.imageInfo} handleClose={this.dockerClosed} docker={this.props.SingleGridCellData.artifacts.docker} /> : null
+                }
+              </div>
+              <div>
+                <IconButton onClick={this.cellClick.bind(this)}>
```

Comment:
Why did we have to bind here and not use the `cellClick = () =>{` syntax in the method?

Reply:
It was not for any reason, just some legacy code from the component I originally forked. Fixed.